braindecode/models/signal_jepa.py

```diff
@@ -0,0 +1,994 @@
+
+    def __init__(
+        self,
+        spat_dim: int,
+        time_dim: int,
+        ch_names: list[str],
+        ch_locs: list[list[float]],
+        sfreq_features: float,
+        spat_kwargs: dict | None = None,
+    ):
+        assert len(ch_names) == len(ch_locs)
+        super().__init__()
+        spat_kwargs = spat_kwargs or {}
+        ch_locs_plus_ukn = [None] + list(ch_locs)
+        self.ch_names = ch_names
+        self.pos_encoder_spat = ChannelEmbedding(
+            ch_locs_plus_ukn, spat_dim, **spat_kwargs
+        )  # (batch_size, n_channels, spat_dim)
+        self.spat_dim = spat_dim
+        self.time_dim = time_dim
+        self.max_n_times = int(self.max_seconds * sfreq_features)
+        self.encoding_time = torch.zeros(0, dtype=torch.float32, requires_grad=False)
+
+    def _check_encoding_time(self, n_times):
+        if self.encoding_time.size(0) < n_times:
+            self.encoding_time = self.encoding_time.new_empty((n_times, self.time_dim))
+            self.encoding_time[:] = pos_encode_time(
+                n_times=n_times,
+                n_dim=self.time_dim,
+                max_n_times=self.max_n_times,
+                device=self.encoding_time.device,
+            )
+
+    def forward(self, batch):
+        """
+        Parameters
+        ----------
+            batch: dict with keys:
+
+                * local_features: (batch_size, n_chans * n_times_out, emb_dim)
+                * ch_idxs: (batch_size, n_chans)
+                    Indices of the channels to use in the ``ch_names`` list passed
+                    as argument plus one. Index 0 is reserved for an unknown channel.
+                    Only needed if ``set_fixed_ch_names`` has not been called.
+
+        Returns
+        -------
+            pos_encoding: (batch_size, n_chans * n_times_out, emb_dim)
+                The first ``spat_dim`` dimensions encode the channels positional encoding
+                and the following ``time_dim`` dimensions encode the temporal positional encoding.
+        """
+        local_features = batch["local_features"]
+        ch_idxs = self.get_ch_idxs(batch).to(local_features.device)
+        batch_size, n_chans_times, emb_dim = local_features.shape
+        batch_size_chs, n_chans = ch_idxs.shape  # ==(1,_) if fixed_ch_names
+        assert emb_dim >= self.spat_dim + self.time_dim
+        assert n_chans_times % n_chans == 0
+        n_times = n_chans_times // n_chans
+
+        pos_encoding = local_features.new_empty(
+            (batch_size_chs, n_chans, n_times, emb_dim)
+        )
+        # Channel pos. encoding
+        pos_encoding[:, :, :, : self.spat_dim] = self.pos_encoder_spat(ch_idxs)[
+            :, :, None, :
+        ]
+        # Temporal pos. encoding
+        self._check_encoding_time(n_times)
+        _ = pos_encoding[:, :, :, self.spat_dim : self.spat_dim + self.time_dim].copy_(
+            self.encoding_time[None, None, :n_times, :],
+        )
```

Comment:
```suggestion
        _ = pos_encoding[:, :, :, self.spat_dim : self.spat_dim + self.time_dim].copy_(
            self.encoding_time[None, None, :n_times, :],
        )
```

Why do we need copy here?
```suggestion
        _ = pos_encoding[:, :, :, self.spat_dim : self.spat_dim + self.time_dim].copy_(
            self.encoding_time[None, None, :n_times, :],
        )
```

Reply:
I don't remember exactly why but I needed it